src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -796,6 +798,38 @@ public function setMaxAge(int $value): object
         return $this;
     }
 
+    /**
+     * Sets the number of seconds after which the response should no longer be returned by shared caches when backend is down.
+     *
+     * This methods sets the Cache-Control stale-if-error directive.
+     *
+     * @return $this
+     *
+     * @final
+     */
+    public function setStaleIfError(int $value): object
```

Comment:
maybe we don't need the new methods?
ppl that want to use the new directive can always use ` $response->setCache(['stale-if-error' => $value]);`
(same for the other method)

Reply:
We cannot do that because setCache method for unknown headers only work for boolean attributes.
That's why I added setters in my PullRequest.
To be more precise it will be handle by this piece of code:

``` PHP
foreach (self::HTTP_RESPONSE_CACHE_CONTROL_DIRECTIVES as $directive => $hasValue) {
            if (!$hasValue && isset($options[$directive])) {
                if ($options[$directive]) {
                    $this->headers->addCacheControlDirective(str_replace('_', '-', $directive));
                } else {
                    $this->headers->removeCacheControlDirective(str_replace('_', '-', $directive));
                }
            }
        }
  ```
Where 
``` PHP
addCacheControlDirective() 
```
is taking a boolean as a parameter